linera-views/src/scylla_db.rs

```diff
@@ -164,34 +164,81 @@ impl ScyllaDbClient {
             return Ok(Vec::new());
         }
         let session = &self.session;
+        let mut map = HashMap::<Vec<u8>, Vec<usize>>::new();
+        let mut unique_keys = Vec::new();
+        for (i_key, key) in keys.into_iter().enumerate() {
+            ensure!(key.len() <= MAX_KEY_SIZE, ScyllaDbStoreError::KeyTooLong);
+            match map.entry(key.clone()) {
+                Entry::Occupied(entry) => {
+                    let entry = entry.into_mut();
+                    entry.push(i_key);
+                }
+                Entry::Vacant(entry) => {
+                    entry.insert(vec![i_key]);
+                    unique_keys.push(key);
+                }
+            }
+        }
```

Comment:
Are duplicate keys common enough to do this optimization? Or does ScyllaDB not allow duplicate keys?

Reply:
By the design of the ScyllaDb operation "k IN (?,...,?)", the test is about key membership.
That design forces the construction of a `HashSet<Vec<u8>, Vec<usize>>` for the indices.

The PR corrects a small error in the construction. It was giving correct results but was slightly inefficient.

The alternative could have been to forbid duplicate keys in the request. But that would be a quite onerous requirement on the design which I think would have been a bad idea.